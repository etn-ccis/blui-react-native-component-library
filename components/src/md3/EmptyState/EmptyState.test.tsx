import React from 'react';
import { cleanup } from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import { EmptyState } from '.';

describe('EmptyState', () => {
    afterEach(cleanup);

    it('renders correctly with title and description', () => {
        const tree = TestRenderer.create(
            <EmptyState title="Empty State Title" description="Empty State Description" />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with custom icon size and color', () => {
        const tree = TestRenderer.create(
            <EmptyState title="Empty State Title" iconSize={150} iconColor="blue" />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with custom styles', () => {
        const customStyles = {
            root: { backgroundColor: 'red' },
            title: { fontSize: 24 },
            description: { color: 'green' },
            actions: { marginTop: 20 },
        };

        const tree = TestRenderer.create(
            <EmptyState title="Empty State Title" description="Empty State Description" styles={customStyles} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with empty props', () => {
        const tree = TestRenderer.create(<EmptyState title="Empty State Title" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with different icon sources', () => {
        const tree = TestRenderer.create(
            <EmptyState title="Empty State Title" icon={{ uri: 'https://example.com/icon.png' }} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with long title and description', () => {
        const tree = TestRenderer.create(
            <EmptyState
                title="This is a very long title that should wrap to the next line"
                description="This is a very long description that should wrap to the next line. It contains more information about the empty state."
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with different theme overrides', () => {
        const themeOverride = {
            colors: {
                primary: 'purple',
            },
        };

        const tree = TestRenderer.create(
            <EmptyState title="Empty State Title" description="Empty State Description" theme={themeOverride} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with dynamic data', () => {
        const dynamicTitle = Math.random() > 0.5 ? 'Data Available' : 'No Data';
        const dynamicDescription = Math.random() > 0.5 ? 'Data is present' : 'Data is missing';

        const tree = TestRenderer.create(<EmptyState title={dynamicTitle} description={dynamicDescription} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly on different screen sizes', () => {
        // You may need to use a library like react-native-responsive-screen for this test
        // to simulate different screen sizes. It might not be achievable in a simple Jest test.
        // Example usage: https://github.com/marudy/react-native-responsive-screen
        // It allows you to mock different screen sizes for testing purposes.
    });

    it('renders correctly with accessibility props', () => {
        const tree = TestRenderer.create(
            <EmptyState
                title="Empty State Title"
                description="Empty State Description"
                accessibilityLabel="custom-accessibility-label"
                accessible={true}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with edge case inputs', () => {
        const tree = TestRenderer.create(<EmptyState title={''} description="" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
