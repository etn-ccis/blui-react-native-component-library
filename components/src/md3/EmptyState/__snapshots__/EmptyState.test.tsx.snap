// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState renders correctly with accessibility props 1`] = `
<View
  accessibilityLabel="custom-accessibility-label"
  accessible={true}
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 16,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          [
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 22,
              "letterSpacing": 0,
              "marginTop": 32,
              "textAlign": "center",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Empty State Title
  </Text>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "400",
            "letterSpacing": 0.25,
            "lineHeight": 20,
          },
          [
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 14,
              "letterSpacing": 0,
              "textAlign": "center",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Empty State Description
  </Text>
</View>
`;

exports[`EmptyState renders correctly with custom icon size and color 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 16,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          [
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 22,
              "letterSpacing": 0,
              "marginTop": 32,
              "textAlign": "center",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Empty State Title
  </Text>
</View>
`;

exports[`EmptyState renders correctly with custom styles 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 16,
      },
      {
        "backgroundColor": "red",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          [
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 22,
              "letterSpacing": 0,
              "marginTop": 32,
              "textAlign": "center",
            },
            {
              "fontSize": 24,
            },
          ],
        ],
      ]
    }
  >
    Empty State Title
  </Text>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "400",
            "letterSpacing": 0.25,
            "lineHeight": 20,
          },
          [
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 14,
              "letterSpacing": 0,
              "textAlign": "center",
            },
            {
              "color": "green",
            },
          ],
        ],
      ]
    }
  >
    Empty State Description
  </Text>
</View>
`;

exports[`EmptyState renders correctly with different icon sources 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 16,
      },
      undefined,
      undefined,
    ]
  }
>
  <Image
    accessibilityElementsHidden={true}
    importantForAccessibility="no-hide-descendants"
    source={
      {
        "uri": "https://example.com/icon.png",
      }
    }
    style={
      [
        {
          "height": 100,
          "resizeMode": "contain",
          "tintColor": "rgba(202, 196, 208, 1)",
          "width": 100,
        },
      ]
    }
  />
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          [
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 22,
              "letterSpacing": 0,
              "marginTop": 32,
              "textAlign": "center",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Empty State Title
  </Text>
</View>
`;

exports[`EmptyState renders correctly with different theme overrides 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 16,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          [
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 22,
              "letterSpacing": 0,
              "marginTop": 32,
              "textAlign": "center",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Empty State Title
  </Text>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "400",
            "letterSpacing": 0.25,
            "lineHeight": 20,
          },
          [
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 14,
              "letterSpacing": 0,
              "textAlign": "center",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Empty State Description
  </Text>
</View>
`;

exports[`EmptyState renders correctly with dynamic data 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 16,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          [
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 22,
              "letterSpacing": 0,
              "marginTop": 32,
              "textAlign": "center",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Data Available
  </Text>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "400",
            "letterSpacing": 0.25,
            "lineHeight": 20,
          },
          [
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 14,
              "letterSpacing": 0,
              "textAlign": "center",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Data is missing
  </Text>
</View>
`;

exports[`EmptyState renders correctly with edge case inputs 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 16,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          [
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 22,
              "letterSpacing": 0,
              "marginTop": 32,
              "textAlign": "center",
            },
            undefined,
          ],
        ],
      ]
    }
  />
</View>
`;

exports[`EmptyState renders correctly with empty props 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 16,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          [
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 22,
              "letterSpacing": 0,
              "marginTop": 32,
              "textAlign": "center",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Empty State Title
  </Text>
</View>
`;

exports[`EmptyState renders correctly with long title and description 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 16,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          [
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 22,
              "letterSpacing": 0,
              "marginTop": 32,
              "textAlign": "center",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    This is a very long title that should wrap to the next line
  </Text>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "400",
            "letterSpacing": 0.25,
            "lineHeight": 20,
          },
          [
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 14,
              "letterSpacing": 0,
              "textAlign": "center",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    This is a very long description that should wrap to the next line. It contains more information about the empty state.
  </Text>
</View>
`;

exports[`EmptyState renders correctly with title and description 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 16,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          [
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 22,
              "letterSpacing": 0,
              "marginTop": 32,
              "textAlign": "center",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Empty State Title
  </Text>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "400",
            "letterSpacing": 0.25,
            "lineHeight": 20,
          },
          [
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 14,
              "letterSpacing": 0,
              "textAlign": "center",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Empty State Description
  </Text>
</View>
`;
